src/cmt_decode_prometheus_remote_write.c

```diff
@@ -0,0 +1,681 @@
+            }
+            histogram->buckets = cmt_histogram_buckets_create_size(spans,
+                                                                   hist->n_positive_spans);
+            free(spans);
+        }
+
+        if (histogram->buckets == NULL) {
+            return CMT_DECODE_PROMETHEUS_REMOTE_WRITE_ALLOCATION_ERROR;
+        }
+    }
+
+    if (n_histograms == 0) {
+        if (map->metric_static_set == CMT_FALSE) {
+            static_metric_detected = CMT_TRUE;
+        }
+    }
+
+    if (static_metric_detected == CMT_FALSE) {
+        metric = calloc(1, sizeof(struct cmt_metric));
+
+        if (metric == NULL) {
+            return CMT_DECODE_PROMETHEUS_REMOTE_WRITE_ALLOCATION_ERROR;
+        }
+
+        cfl_list_init(&metric->labels);
+
+        result = decode_labels(cmt,
+                               map,
+                               metric,
+                               n_labels,
+                               labels);
+
+        if (result) {
+            destroy_label_list(&metric->labels);
+
+            free(metric);
+        }
+        else {
+            cfl_list_add(&metric->_head, &map->metrics);
+        }
+    }
+    else {
+        metric = &map->metric;
+
+        map->metric_static_set = CMT_TRUE;
+    }
+
+    if (result == CMT_DECODE_PROMETHEUS_REMOTE_WRITE_SUCCESS) {
+        if (hist->n_negative_spans > 0) {
+            for (i = 0; i < hist->n_negative_counts; i++) {
+                cmt_metric_hist_set(metric, hist->timestamp * 1000000,
+                                    i, hist->negative_counts[i]);
+            }
+        }
+        else if (hist->n_positive_spans > 0) {
+            for (i = 0; i < hist->n_positive_counts; i++) {
+                cmt_metric_hist_set(metric, hist->timestamp * 1000000,
+                                    i, hist->positive_counts[i]);
+            }
+        }
+        else {
+            free(metric);
+
+            return CMT_DECODE_PROMETHEUS_REMOTE_WRITE_DECODE_ERROR;
+        }
+    }
+
+    metric->hist_sum = cmt_math_d64_to_uint64(hist->sum);
+    if (hist->count_case == PROMETHEUS__HISTOGRAM__COUNT_COUNT_INT) {
+        metric->hist_count = hist->count_int;
+    }
+    else if (hist->count_case == PROMETHEUS__HISTOGRAM__COUNT_COUNT_FLOAT) {
+        metric->hist_count = hist->count_float;
+    }
+    else {
+        free(metric);
+
+        return CMT_DECODE_PROMETHEUS_REMOTE_WRITE_DECODE_ERROR;
+    }
+
+    return result;
+}
+
+static int decode_histogram_time_series(struct cmt *cmt,
+                                        struct cmt_map *map,
+                                        size_t hist_count,
+                                        Prometheus__TimeSeries *ts)
+{
+    size_t index;
+    int    result;
+    Prometheus__Histogram *histogram = NULL;
+
+    result = CMT_DECODE_PROMETHEUS_REMOTE_WRITE_SUCCESS;
+
+    for (index = 0 ;
+         result == 0 &&
+         index < hist_count ; index++) {
```

Comment:
this and previous line fits fine in the first conditional line

Reply:
Got it. Fixed in https://github.com/fluent/cmetrics/pull/201/commits/8d6880d91702bd89b46804e1807a323d7d90102e.